client/src/pages/LoginPage.jsx

```diff
@@ -1,79 +1,83 @@
-import {useState} from 'react'
+import { useState, useContext } from 'react';
 import { useNavigate } from 'react-router-dom';
-import { UsersContext } from '../App';
-import { useContext } from 'react';
+import { loginUser } from '../services/apiService';
+import { UserContext } from '../App';
 
-function LoginPage () {
-  const basedUrl = 'http://localhost:3000/'
-  const navigate = useNavigate()
-
-  const {userLogged, setUserLogged} = useContext(UsersContext)
-  
+function LoginPage() {
+  const { setUser } = useContext(UserContext)
   const [login, setLogin] = useState({
     usernameOrEmail: '',
     password: ''
   })
 
+  const navigate = useNavigate()
+
   const handleChange = (event) => {
-    const {name, value} = event.target
+    const { name, value } = event.target
 
     setLogin({
       ...login,
       [name]: value
     });
   };
 
-const handleSubmit = async (event) => {
-  event.preventDefault();
+  const handleSubmit = async (event) => {
+    event.preventDefault();
+    const { usernameOrEmail, password } = login;
 
-  const { usernameOrEmail, password } = login;
+    try {
+      const response = await loginUser({ usernameOrEmail, password });
 
-  try {
-    const response = await fetch(basedUrl+'login', {
-      method: 'POST',
-      headers: {
-        'Content-Type': 'application/json',
-      },
-      body: JSON.stringify({
-        usernameOrEmail,
-        password,
-      }),
-    });
-
-    if (response.ok) {
-      const user = await response.json();
-      console.log(user)
-      console.log('Logged in user:', user); //Navigate to userPage!! mayin the response of the request include the id to use it as param 
-      navigate(`/`, {replace: true})
-      setUserLogged(user.username)
-    } else {
-      alert('Invalid username/email or password.')
+      if (response.ok) {
+        const user = await response.json();
+        console.log('Logged in user:', user); //Navigate to userPage!! mayin the response of the request include the id to use it as param 
+        navigate(`/`, { replace: true })
+        setUser(user)

```

Comment:
It will not help, to change the order:  calling setUser() first does not fully prevent the "unmounted component state update" issue. It reduces the chance of encountering it, but doesn't guarantee safety because setUser() is asynchronous.
But
Good point,
We can do check like isMounted(componentRef) ... but its a bit of overkill at this point

Reply:
but will anyways, thanks!